src/platforms/esp32/components/avm_builtins/spi_driver.c

```diff
@@ -54,6 +54,17 @@
 
 #define TAG "spi_driver"
 
+#ifdef CONFIG_IDF_TARGET_ESP32S2
+#define VSPI_HOST   SPI2_HOST
+#define HSPI_HOST   SPI3_HOST
+#elif CONFIG_IDF_TARGET_ESP32S3
+#define VSPI_HOST   SPI2_HOST
+#define HSPI_HOST   SPI3_HOST
+#elif CONFIG_IDF_TARGET_ESP32C3
+// only one user SPI bus, no VSPI
+#define HSPI_HOST   SPI2_HOST
+#endif
```

Comment:
so, basically HSPI and VSPI are not defined on newer chips, right?

Reply:
Correct. For some reason Espressif does not define HSPI or VSPI for the newer chips. SPI#_HOST are always available for all of the chipsets, so we may want to move to using that naming convention rather than HSPI and VSPI, but that is likely an issue we will want to discuss as a separate topic.